parsl/executors/taskvine/executor.py

```diff
@@ -473,68 +371,88 @@ def submit(self, func, resource_specification, *args, **kwargs):
-        map_file = self._path_in_task(executor_task_id, "map")
+            logger.debug("Creating executor task {} with function at: {} and result to be found at: {}".format(executor_task_id, function_file, result_file))
 
-        logger.debug("Creating Task {} with function at: {}".format(executor_task_id, function_file))
-        logger.debug("Creating Executor Task {} with result to be found at: {}".format(executor_task_id, result_file))
+            # Pickle the result into object to pass into message buffer
+            self._serialize_function(call_specs['app_type'], function_file, func, args, kwargs)
 
-        self._serialize_function(function_file, func, args, kwargs)
+            # Construct the map file of local filenames at worker
+            self._construct_map_file(map_file, input_files, output_files)
 
-        if self.pack:
+        # Register a tarball containing all package dependencies for this app if instructed
+        if self.manager_config.app_pack:
             env_pkg = self._prepare_package(func, self.extra_pkgs)
         else:
             env_pkg = None
 
-        if self.pack_env:
-            env_pkg = self.pack_env
-
-        logger.debug("Constructing map for local filenames at worker for task {}".format(executor_task_id))
-        self._construct_map_file(map_file, input_files, output_files)
-
         if not self.submit_process.is_alive():
             raise ExecutorError(self, "taskvine Submit Process is not alive")
 
         # Create message to put into the message queue
         logger.debug("Placing task {} on message queue".format(executor_task_id))
+
+        # Put apps into their categories based on function name if enabled
         if category is None:
-            category = func.__name__ if self.autocategory else 'parsl-default'
-        self.task_queue.put_nowait(ParslTaskToVine(executor_task_id,
-                                                   category,
-                                                   cores,
-                                                   memory,
-                                                   disk,
-                                                   gpus,
-                                                   priority,
-                                                   running_time_min,
-                                                   env_pkg,
-                                                   map_file,
-                                                   function_file,
-                                                   result_file,
-                                                   input_files,
-                                                   output_files))
+            category = func.__name__ if self.manager_config.autocategory else 'parsl-default'
+
+        # need source of function to send tasks as python or serverless tasks
+        if exec_mode == 'python' or exec_mode == 'serverless':
+            func_src = autopep8.fix_code(inspect.getsource(func))
```

Comment:
does this formatting make the code more executable? or it is prettifying for human consumption? (i can't tell from the autopep8 docs if this is going to change how the code runs?)

Reply:
In cases where apps are functions indented inside a class or another function, like this:
```
def f():
    @python_app
    def g():
        ...
```,
then inspecting function g `inspect.getsource(g)` results in a string that's 1 level indented, like this:
```
@python_app
    def g():
        ...
```
This prevents attempts to exec the source code into a temporary namespace as the indentation level is one level to the right. `autopep8.fix_code` corrects this and aligns the function with its body properly.

Another way to address this case is to write a custom method that aligns function source code (detecting type, either tabs or whitespaces, and number of indentations should be easy) but this is a bit hacky. Let me know what you think.

Reply:
mostly I think this PR should avoid anything to do with changing serialization away from parsl's defaults, and focus on solid improvements to the TaskVineExecutor.

Then, work on serialization in a different branch - it might even be that some techniques like are more generally useful across all parsl executors and the code should go into the parsl serialization system, rather than the taskvine executor.